pages/treeherder.py

```diff
@@ -52,6 +52,81 @@ def open_perfherder_page(self):
         return PerfherderPage(self.base_url, self.selenium).wait_for_page_to_load()
 
 
+    class Result(PageRegion):
+
+        _job_result_status_locator = (By.CSS_SELECTOR, '#result-status-pane > div:nth-child(1) > span')
+        _result_set_locator = (By.CSS_SELECTOR, '#th-global-content .result-set')
+        _selected_job_title_locator = (By.CSS_SELECTOR, '.job-list .selected-job')
+
+        @property
+        def job_result_status(self):
+            return self.selenium.find_element(*self._job_result_status_locator).text
+
+        @property
+        def result(self):
+            return [Result(self, root=el) for el in
+                self.find_elements(self._result_set_locator)]
+
+        def pin_using_spacebar(self):
+            el = self.selenium.find_element(*self._result_set_locator)
+            Wait(self.selenium, self.timeout).until(EC.visibility_of(el))
+            el.send_keys(Keys.SPACE)
+            Wait(self.selenium, self.timeout).until(lambda s: self.pinboard.is_pinboard_open)
+
+        def select_next_job(self):
+            el = self.selenium.find_element(*self._result_set_locator)
+            Wait(self.selenium, self.timeout).until(EC.visibility_of(el))
+            el.send_keys(Keys.ARROW_RIGHT)
+            Wait(self.selenium, self.timeout).until(lambda s: self.job_result_status)
+            return self.selenium.find_element(*self._selected_job_title_locator).get_attribute('title')
+
+    class JobDetails(PageRegion):
```

Comment:
Hmm.. I see that all locators are already using IDs, and you're using Selenium's expected conditions, which locate from the document root, so perhaps this can be left as it is.